iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalCatalog.java

```diff
@@ -183,6 +185,43 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
     transaction.commitTransaction();
   }
 
+  public Page<SoftDeletedTableDto> searchSoftDeletedTables(
+      Namespace namespace, String tableId, Pageable pageable) {
+    if (namespace.levels().length > 1) {
+      throw new ValidationException(
+          "Input namespace has more than one levels " + String.join(".", namespace.levels()));
+    }
+
+    try {
+      List<SoftDeletedTableDto> softDeletedTables =
+          houseTableRepository.searchSoftDeletedTables(namespace.toString(), tableId, pageable)
+              .stream()
+              .map(
+                  houseTable ->
+                      SoftDeletedTableDto.builder()
+                          .tableId(houseTable.getTableId())
+                          .databaseId(houseTable.getDatabaseId())
+                          .tableLocation(houseTable.getTableLocation())
+                          .deletedAtMs(houseTable.getDeletedAtMs())
+                          .build())
+              .collect(Collectors.toList());
+
+      return new PageImpl<>(softDeletedTables, pageable, softDeletedTables.size());
```

Comment:
You don't need to convert the page to list and then back to page. Just use `houseTableRepository.searchSoftDeletedTables().map()`.

Reply:
Yes +1.